src/test/java/com/inventorsoft/junit/mapper/BookMapperTest.java

```diff
@@ -0,0 +1,185 @@
+        BookResponse givenResult = bookMapper.mapEntityToResponse(book);
+
+        assertEquals(bookResponse.getId(), givenResult.getId());
+        assertEquals(bookResponse.getTitle(), givenResult.getTitle());
+        assertEquals(bookResponse.getReleaseDate(), givenResult.getReleaseDate());
+        assertEquals(bookResponse.getDescription(), givenResult.getDescription());
+        assertEquals(bookResponse.getAuthor(), givenResult.getAuthor());
+    }
+
+    @Test
+    public void mapEntitiesToResponses() {
+        List<Book> books = List.of(book);
+        List<BookResponse> expectedResults = List.of(bookResponse);
+
+        when(authorMapper.mapEntityToResponse(book.getAuthor())).thenReturn(authorResponse);
+        List<BookResponse> givenResults = bookMapper.mapEntitiesToResponses(books);
+
+        assertEquals(expectedResults.size(), givenResults.size());
+        IntStream.range(0, expectedResults.size()).forEachOrdered(i -> {
+            assertEquals(expectedResults.get(i).getId(), givenResults.get(i).getId());
+            assertEquals(expectedResults.get(i).getDescription(), givenResults.get(i).getDescription());
+            assertEquals(expectedResults.get(i).getReleaseDate(), givenResults.get(i).getReleaseDate());
+            assertEquals(expectedResults.get(i).getTitle(), givenResults.get(i).getTitle());
+            assertEquals(expectedResults.get(i).getAuthor().getId(), givenResults.get(i).getAuthor().getId());
+            assertEquals(expectedResults.get(i).getAuthor().getName(), givenResults.get(i).getAuthor().getName());
+            assertEquals(expectedResults.get(i).getAuthor().getBirthDate(), givenResults.get(i).getAuthor().getBirthDate());
+        });
+    }
+
+    @Test
+    public void mapCreateRequestToEntity() {
+        CreateBookRequest createBookRequest = new CreateBookRequest();
+        createBookRequest.setTitle(BOOK_TITLE);
+        createBookRequest.setDescription(BOOK_DESCRIPTION);
+        createBookRequest.setReleaseDate(BOOK_RELEASE_DATE);
+        createBookRequest.setAuthorId(AUTHOR_ID);
+
+        when(authorRepository.findById(createBookRequest.getAuthorId())).thenReturn(Optional.of(author));
+        Book givenResult = bookMapper.mapCreateRequestToEntity(createBookRequest);
+
+        assertEquals(book.getId(), givenResult.getId());
+        assertEquals(book.getTitle(), givenResult.getTitle());
+        assertEquals(book.getDescription(), givenResult.getDescription());
+        assertEquals(book.getReleaseDate(), givenResult.getReleaseDate());
+        assertEquals(book.getAuthor(), givenResult.getAuthor());
+    }
+
+    @Test
+    public void whenAuthorIdIsNull_thenExceptionIsThrown() {
+        assertThrows(RuntimeException.class,
+                () -> authorRepository.findById(null)
+                        .orElseThrow(RuntimeException::new));
+    }
+
+    @Test
+    public void mapCreateRequestToEntityShouldThrowNullPointExceptionWhenRequestIsNull() {
```

Comment:
good one, but there is no logic inside `mapCreateRequestToEntity` method, so it is not needed and will not add code coverage